src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -152,6 +173,30 @@ public Float getIndexSizeAsPercentage(final String indexName) {
         return 100 * getIndexSizeInKilobytes(indexName) / (float) KNNSettings.getCircuitBreakerLimit().getKb();
     }
 
+    /**
+     * Getter for current size of all training jobs in Kilobytes.
+     *
+     * @return current size of the cache
+     */
+    public long getTrainingSizeInKilobytes() {
+        // Currently, all allocations that are not index allocations will be for training.
+        return cache.asMap().values().stream()
+                .filter(nativeMemoryAllocation ->
+                        nativeMemoryAllocation instanceof NativeMemoryAllocation.TrainingDataAllocation ||
+                                nativeMemoryAllocation instanceof NativeMemoryAllocation.AnonymousAllocation)
+                .mapToLong(NativeMemoryAllocation::getSizeInKB)
+                .sum();
+    }
+
+    /**
+     * Returns how full the cache is as a percentage of the total cache capacity.
+     *
+     * @return Percentage of the cache full
+     */
+    public Float getTrainingSizeAsPercentage() {
+        return 100 * getTrainingSizeInKilobytes() / (float) KNNSettings.getCircuitBreakerLimit().getKb();
```

Comment:
Do we have net to avoid divide by zero error?

Reply:
Divide by 0 produces NaN:
```
      "graph_memory_usage_percentage" : "NaN",
      "graph_query_requests" : 0,
      "graph_memory_usage" : 0,
      "cache_capacity_reached" : false,
      "load_success_count" : 0,
      "training_memory_usage" : 0,
      "indices_in_cache" : { },
      "script_query_errors" : 0,
      "hit_count" : 0,
      "knn_query_requests" : 0,
      "total_load_time" : 0,
      "miss_count" : 0,
      "training_memory_usage_percentage" : "NaN",
      "graph_index_requests" : 0,
      "faiss_initialized" : false,
      "load_exception_count" : 0,
      "training_errors" : 0,
      "eviction_count" : 0,
      "nmslib_initialized" : false,
      "script_compilations" : 0,
      "script_query_requests" : 0,
      "graph_query_errors" : 0,
      "indexing_from_model_degraded" : false,
      "graph_index_errors" : 0,
      "training_requests" : 0,
      "script_compilation_errors" : 0
```

For stats, I think this is okay - Im not sure what else we would be a better output. What are your thoughts @VijayanB ?

Reply:
i prefer 0 than NaN